L1Trigger/L1THGCalUtilities/python/caloTruthCells_cff.py

```diff
@@ -1,70 +1,89 @@
-    hgcalTruthBackEndLayer2Producer *
-    hgcalTruthTowerMapProducer *
-    hgcalTruthTowerProducer
-)
+# If caloTruthCellsProducer.makeCellsCollection is True, can run the clustering algorithm over output cells too
 
-from L1Trigger.L1THGCalUtilities.hgcalTriggerNtuples_cfi import *
+if caloTruthCellsProducer.makeCellsCollection:
+    ## cluster and tower sequence
 
-ntuple_triggercells.caloParticlesToCells = cms.InputTag('caloTruthCellsProducer')
-ntuple_triggercells.FillTruthMap = cms.bool(True)
+    from L1Trigger.L1THGCal.hgcalTriggerGeometryESProducer_cfi import *
+    from L1Trigger.L1THGCal.hgcalConcentrator_cff import *
+    from L1Trigger.L1THGCal.hgcalBackEndLayer1_cff import *
+    from L1Trigger.L1THGCal.hgcalBackEndLayer2_cff import *
+    from L1Trigger.L1THGCal.hgcalTowerMap_cff import *
+    from L1Trigger.L1THGCal.hgcalTower_cff import *
+    
+    hgcalTruthConcentratorProducer = hgcalConcentratorProducer.clone(
+        InputTriggerCells = cms.InputTag('caloTruthCellsProducer:HGCalVFEProcessorSums')
+    )
+    
+    hgcalTruthBackEndLayer1Producer = hgcalBackEndLayer1Producer.clone(
+        InputTriggerCells = cms.InputTag('hgcalTruthConcentratorProducer:HGCalConcentratorProcessorSelection')
+    )
+    
+    hgcalTruthBackEndLayer2Producer = hgcalBackEndLayer2Producer.clone(
+        InputCluster = cms.InputTag('hgcalTruthBackEndLayer1Producer:HGCalBackendLayer1Processor2DClustering')
+    )
+    
+    hgcalTruthTowerMapProducer = hgcalTowerMapProducer.clone(
+        InputTriggerCells = cms.InputTag('caloTruthCellsProducer:HGCalVFEProcessorSums')
+    )
+    
+    hgcalTruthTowerProducer = hgcalTowerProducer.clone(
+        InputTowerMaps = cms.InputTag('hgcalTruthTowerMapProducer:HGCalTowerMapProcessor')
+    )
+    
+    caloTruthCells += cms.Sequence(
+        hgcalTruthConcentratorProducer *
+        hgcalTruthBackEndLayer1Producer *
+        hgcalTruthBackEndLayer2Producer *
+        hgcalTruthTowerMapProducer *
+        hgcalTruthTowerProducer
+    )
 
-ntuple_triggercells_truth = ntuple_triggercells.clone(
-    TriggerCells = cms.InputTag('caloTruthCellsProducer'),
-    MultiClusters = cms.InputTag('hgcalTruthBackEndLayer2Producer:HGCalBackendLayer1Processor2DClustering'),
-    Prefix = cms.untracked.string('tctruth'),
-    FillTruthMap = cms.bool(False)
-)
+    ## ntuplize clusters and towers
 
-ntuple_clusters_truth = ntuple_clusters.clone(
-    Clusters = cms.InputTag('hgcalTruthBackEndLayer1Producer:HGCalBackendLayer1Processor2DClustering'),
-    Prefix = cms.untracked.string('cltruth')
-)
+    ntuple_triggercells.caloParticlesToCells = cms.InputTag('caloTruthCellsProducer')
+    ntuple_triggercells.FillTruthMap = cms.bool(True)
+    
+    ntuple_triggercells_truth = ntuple_triggercells.clone(
+        TriggerCells = cms.InputTag('caloTruthCellsProducer:HGCalVFEProcessorSums'),
+        MultiClusters = cms.InputTag('hgcalTruthBackEndLayer2Producer:HGCalBackendLayer1Processor2DClustering'),
```

Comment:
I believe there is a typo here, it should actually be `Multiclusters`

Reply:
Indeed!